Api/API Standardization.md

```diff
@@ -0,0 +1,305 @@
+
+> *The early Web architecture defined URI as document identifiers. Authors were instructed to define identifiers in terms of a document’s location on the network. Web protocols could then be used to retrieve that document. However, this definition proved to be unsatisfactory for a number of reasons. First, it suggests that the author is identifying the content transferred, which would imply that the identifier should change whenever the content changes. Second, there exist many addresses that corresponded to a service rather than a document — authors may be intending to direct readers to that service, rather than to any specific result from a prior access of that service. Finally, there exist addresses that do not correspond to a document at some periods of time, such as when the document does not yet exist or when the address is being used solely for naming, rather than locating information.*
+> 
+
+This led to the redefinition of the term resource in the scope of HTTP and URI:
+
+> *Defining* resource *such that a URI identifies a concept rather than a document leaves us with another question: how does a user access, manipulate, or transfer a concept such that they can get something useful when a hypertext link is selected? REST answers that question by defining the things that are manipulated to be representations of the identified resource, rather than the resource itself. An origin server maintains a mapping from resource identifiers to the set of representations corresponding to each resource. A resource is therefore manipulated by transferring representations through the generic interface defined by the resource identifier.*
+> 
+
+This redefinition demonstrates the importance of the distinction between what is served and accepted by the API (a resource representation) and what the API really works with in the background (a resource). When building an API, designing resource representations should focus on what is useful to the clients that will consume the API. Regardless if we’re building something as specific as building a backend for frontend (BFF) or as generic as a public-facing API, the outermost layer’s design must be led with the question “How will this API be used?”. In contrast, the resource representation should not greatly influence the resources themselves - we don’t want to impose unreasonable restrictions on our DB schema, resource relations and processes for the convenience of the API consumers. Same logic can be applied the other way round: the way we manipulate the resources on the backend shouldn’t make consuming the API more complex than the process we’re implementing requires it to be. The backend owns the mapping of resource representations to the resources, which gives it the liberty to be creative and enable both sides to work as simple and efficiently as possible.
+
+# URL standard
+
+## Casing
+
+All parts of an URL are written in `camelCase`.
```

Comment:
for me it is more readable if we use kebab-case for the endpoint, and camelCase for the query parameters, but I will adapt to the case the majority agrees upon 👍

Reply:
I agree with this comment. Same preference for kebab-case, but definitely wouldn't aim for introducing something not .NET default for query parameters.